packages/itwinui-react/src/core/LabeledSelect/LabeledSelect.types-test.tsx

```diff
@@ -20,7 +20,7 @@ import { LabeledSelect } from './LabeledSelect.js';
           { value: 2, label: 'Option 2' },
           { value: 3, label: 'Option 3' },
         ]}
-        onChange={(value) => {
+        onChange={(value: number) => {
```

Comment:
Why is this necessary? It should be automatically inferred.

Reply:
Similar to https://github.com/iTwin/iTwinUI/pull/1886#discussion_r1516824519, doing `Omit<SelectProps<T>, "styleType">` instead of `SelectProps<T>` gave strange type behaviors and errors. That comment was the resulting error. This comment is the resulting strange behavior.

I made a mention of this case in the GitHub gist that I had included in the PR description. I tried quite a few approaches to fix the "potential issues" I mentioned there, but I couldn't fix the last remaining ones, which included this issue of `onChange`'s `value` argument not getting the inferred type. Open to different approaches.

Reply:
Looks like `styleType?: never` works as a workaround.

Reply:
Yes, after our recent commits (https://github.com/iTwin/iTwinUI/pull/1886/files/ae07a6fb42b4d464e50d42a3cd0347dc31c43da0..aca5cf10a4c6b7504e9eae54b630913074789f41), that highlighted change was no longer needed, and thus reverted.